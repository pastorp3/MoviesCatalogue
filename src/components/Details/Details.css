.details-poster {
  width: 300px;
  height: 450px;
  border-radius: 10px;
}

.main-container-details {
  padding: 30px 50px 30px 50px;
  background-image: linear-gradient(to right, rgba(8.63%, 8.24%, 8.63%, 1) 150px, rgba(8.63%, 8.24%, 8.63%, 0.84) 100%);
  display: flex;
  font-family: sans-serif;
}

.poster-background {
  background-size: cover;
  background-repeat: no-repeat;
  background-position: right -200px top;
}

.details-title {
  font-size: 2.2rem;
  font-weight: bold;
}

.details-text {
  padding: 30px 0 30px 50px;
  color: white;
}

.details-language {
  margin-left: 5px;
}

.details-dot {
  align-items: center;
  font-weight: bold;
  margin-right: 5px;
  margin-left: 5px;
}

.genre {
  margin-right: 5px;
}

.details-progress {
  width: 110px;
  height: 60px;
  display: flex;
  flex-direction: row;
  font-weight: bold;
  align-items: center;
}

.details-score {
  margin-left: 10px;
}

.details-second-container {
  padding-top: 25px;
  display: flex;
  align-items: center;
  font-weight: bold;
  margin-bottom: 10px;
}

.play-icon {
  width: 80px;
  height: 80px;
  padding-left: 20px;
}

.tagline {
  color: gainsboro;
}

.overview-title {
  margin-top: 10px;
  font-weight: bold;
  margin-bottom: 10px;
}

.details-companies {
  font-weight: bold;
  margin-top: 15px;
  padding: 30px 50px 30px 50px;
  font-family: sans-serif;
}

.details-status {
  font-weight: bold;
  margin-top: 15px;
}

.companies-container {
  display: flex;
  flex-wrap: wrap;
}

.logo-company {
  margin-right: 20px;
  width: 150px;
  height: 120px;
}
