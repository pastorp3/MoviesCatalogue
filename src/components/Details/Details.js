/* eslint-disable  prefer-template, max-len */
import axios from 'axios';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import Navbar from '../Navbar/Navbar';
import './Details.css';
import play from '../../assets/play.png';
import Footer from '../Footer/Footer';

const Details = () => {
  const { id } = useParams();
  const [details, setDetails] = useState(false);
  useEffect(() => {
    axios.get(`https://api.themoviedb.org/3//movie/${id}?api_key=85c461272a40b11f9fdd119183a15d9a`)
      .then(({ data }) => {
        setDetails(data);
      });
  }, []);
  const imgSrc = `https://image.tmdb.org/t/p/w500/${details.poster_path}`;
  const styles = {
    backgroundImage: 'url(https://image.tmdb.org/t/p/w1920_and_h800_multi_faces/' + details.backdrop_path + ')',
  };
  const percentage = details.vote_average * 10;
  return (
    <div>
      <Navbar />
      <div>
        <div className="poster-background" style={styles}>
          <div className="main-container-details">
            <div>
              { details && <img alt="poster-details" src={imgSrc} className="details-poster" /> }
            </div>
            <div className="details-text">
              <span className="details-title">{details.original_title}</span>
              <br />
              <span>{details && details.release_date}</span>
              <span className="details-language">
                (
                {details.original_language}
                )
              </span>
              <span className="details-dot">|</span>
              { details && details.genres.map((genre) => <span className="genre" key={genre.id}>{genre.name}</span>)}
              <span className="details-dot">|</span>
              <span>
                {details && details.runtime}
                m
              </span>
              <br />
              <div className="details-second-container">
                <div className="details-progress">
                  <CircularProgressbar
                    value={percentage}
                    text={`${percentage}%`}
                    styles={buildStyles({
                      textColor: '#fff',
                      pathColor: '#228B22',
                      textSize: '26px',
                    })}
                  />
                  <span className="details-score">User Score</span>
                </div>
                <img src={play} alt="playIcon" className="play-icon" />
                <span>Play Trailer</span>
              </div>
              <div className="tagline">{details && details.tagline}</div>
              <div className="overview-title">Overview</div>
              <div>
                {details && details.overview}
              </div>
              <div className="details-status">
                <span>Status: </span>
                <span>{details && details.status}</span>
              </div>
            </div>
          </div>
        </div>
        <div className="details-companies">
          <h2>Production Companies</h2>
          <div className="companies-container">
            {
              details && details.production_companies.map((company) => <img alt={company.name} className="logo-company" key={company.id} src={`https://image.tmdb.org/t/p/w200/${company.logo_path}`} />)
            }
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
};
export default Details;

/* eslint-enable */
