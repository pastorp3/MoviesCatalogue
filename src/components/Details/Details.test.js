import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import Details from './Details';

describe('Details Component', () => {
  it('Should render according with snapshout whithout error', () => {
    const tree = renderer.create(<MemoryRouter><Details /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
