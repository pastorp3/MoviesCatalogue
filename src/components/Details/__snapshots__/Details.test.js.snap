// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component Should render according with snapshout whithout error 1`] = `
<div
  data-test="Details"
>
  <nav
    data-test="navComponent"
  >
    <img
      alt="Logo"
      className="logo"
      data-test="logo"
      src="logo.png"
    />
    <a
      className="home-link"
      data-test="homeLink"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </nav>
  <div>
    <div
      className="poster-background"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/w1920_and_h800_multi_faces/undefined)",
        }
      }
    >
      <div
        className="main-container-details"
      >
        <div />
        <div
          className="details-text"
        >
          <span
            className="details-title"
          />
          <br />
          <span />
          <span
            className="details-language"
          >
            (
            )
          </span>
          <span
            className="details-dot"
          >
            |
          </span>
          <span
            className="details-dot"
          >
            |
          </span>
          <span>
            m
          </span>
          <br />
          <div
            className="details-second-container"
          >
            <div
              className="details-progress"
            >
              <svg
                className="CircularProgressbar "
                data-test-id="CircularProgressbar"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <path
                  className="CircularProgressbar-trail"
                  d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                  fillOpacity={0}
                  strokeWidth={8}
                  style={
                    Object {
                      "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                      "strokeDashoffset": "0px",
                    }
                  }
                />
                <path
                  className="CircularProgressbar-path"
                  d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                  fillOpacity={0}
                  strokeWidth={8}
                  style={
                    Object {
                      "stroke": "#228B22",
                      "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                      "strokeDashoffset": "NaNpx",
                    }
                  }
                />
                <text
                  className="CircularProgressbar-text"
                  style={
                    Object {
                      "fill": "#fff",
                      "fontSize": "26px",
                    }
                  }
                  x={50}
                  y={50}
                >
                  NaN%
                </text>
              </svg>
              <span
                className="details-score"
              >
                User Score
              </span>
            </div>
            <img
              alt="playIcon"
              className="play-icon"
              src="play.png"
            />
            <span>
              Play Trailer
            </span>
          </div>
          <div
            className="tagline"
          />
          <div
            className="overview-title"
          >
            Overview
          </div>
          <div />
          <div
            className="details-status"
          >
            <span>
              Status: 
            </span>
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      className="details-companies"
    >
      <h2>
        Production Companies
      </h2>
      <div
        className="companies-container"
      />
    </div>
  </div>
  <div
    className="footer"
    data-test="Footer"
  >
    <span>
      Copyright 2021
    </span>
  </div>
</div>
`;
